### PR TITLE
Init/webpack

### PR DESCRIPTION
**Pull request by angelahuang89**
## Webpack setup
- setup webpack dev and prod config
- setup client index.jsx, App.jsx, and index.html with React
- setup server index.js and server.js
- setup dev-middleware and hot-middleware for development using environmental check helper
- setup environmental variables with dotenv

### CODE REVIEW

**Comment by angelahuang89** [comment]
[Set up Webpack boilerplate](https://trello.com/c/ugnGPcE3/36-set-up-webpack-boilerplate)

File: helpers/environmentCheck.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {} from 'dotenv/config';
+
+const checkEnvironment = (app) => {
```
Comment (andino93) [inline helpers/environmentCheck.js]: The name of this file and function don't really correspond with it's function. Perhaps something like `useWebpackMiddleware` would be a more accurate name

**Review by andino93 — CHANGES_REQUESTED** [review]
Looks good! Just a small change and it'll be good to go!👌👍

File: helpers/environmentCheck.js (hunk 1 above)
Comment (angelahuang89) [reply helpers/environmentCheck.js]: Thanks for the suggestion! Renamed the file and function to be `useWebpackMiddleware` for clarity.

**Comment by angelahuang89** [comment]
Cool, thanks for the review! Started looking into minifying for webpack prod config. Is there a preferred minification method?

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import useWebpackMiddleware from '../helpers/useWebpackMiddleware';
```
Comment (andino93) [inline server/server.js:2]: Ah just another small change. This is technically not a helper so it should be a folder with the server called `environment` or `middleware`. Though in this case,  your function isn't a middleware so prob environment is a better choice for this case.

**Review by andino93 — CHANGES_REQUESTED** [review]

**Comment by andino93** [comment]
Look into browerfy and webpack minification. Both would be good I think

File: server/server.js (hunk 2 above)
Comment (angelahuang89) [reply server/server.js:2]: Okay, that makes sense. Moved `useWebpackMiddleware` to server environment folder.

**Comment by angelahuang89** [comment]
Alright, I went with the UglifyjsWebpackPlugin for now.

**Review by andino93 — APPROVED** [review]
Awesome work! =D

**Pull request merged**